pkg/crc/segment/segment_test.go

```diff
@@ -0,0 +1,55 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ConsentTelemetry = "consent-telemetry"
+)
+
+func newTestConfig(configFile string, value bool, envPrefix string) (*crcConfig.Config, error) {
+	storage, err := crcConfig.NewViperStorage(configFile, envPrefix)
+	if err != nil {
+		return nil, err
+	}
+	config := crcConfig.New(storage)
+	config.AddSetting(ConsentTelemetry, value, crcConfig.ValidateBool, crcConfig.SuccessfullyApplied)
+	return config, nil
+}
+
+func TestClientUploadWithoutConcent(t *testing.T) {
+	dir, err := ioutil.TempDir("", "cfg")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+	configFile := filepath.Join(dir, "crc.json")
+
+	config, err := newTestConfig(configFile, false, "CRC")
+	require.NoError(t, err)
+
+	c, err := NewClient(config)
+	require.NoError(t, err)
+	require.NoError(t, c.Upload(nil))
```

Comment:
This is the hard part of the test. You can change the `Endpoint` (https://github.com/segmentio/analytics-go/blob/master/analytics.go#L114) to a embedded http server and count the number of calls received. It should be 0 in this case.

When we will add more data, we can then analyze the payload and write assertions for some fields.

Reply:
Right now I think the only way to perform those unit test is to modify the `Upload` signature to accept the `client`, `identity_file`, `err` message because there is no way I can mock all those bit in current scenario. Then I can create a new dummy client with different endpoint and also pass it to Upload and get the result.